src/WattleScript.Interpreter/Execution/VM/Processor/Processor_InstructionLoop.cs

```diff
@@ -1479,25 +1479,32 @@ private void ExecTblInitI(Instruction i)
 			m_ValueStack.RemoveLast(i.NumVal);
 		}
 
-		private void ExecNewRange()
+		private void ExecNewRange(Instruction i)
 		{
-			DynValue toDv = m_ValueStack.Pop().ToScalar();
-			DynValue fromDv = m_ValueStack.Pop().ToScalar();
+			if (i.NumVal3 == 1) // from v-stack
+			{
+				DynValue toDv = m_ValueStack.Pop().ToScalar();
```

Comment:
This should just be `m_ValueStack.Pop().CastToInt();` instead of the extraneous ToScalar

Reply:
I need `from/toDv.Type` for errors. ToScalar removed

```
DynValue toDv = m_ValueStack.Pop();
DynValue fromDv = m_ValueStack.Pop();

int? fromInt = fromDv.CastToInt();
int? toInt = toDv.CastToInt();
```